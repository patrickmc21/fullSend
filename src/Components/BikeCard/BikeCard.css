.bike-card {
  display: inline-block;
  border: 1px solid #292727;
  border-radius: 13px;
  margin: 0 1.2rem 50px 1.2rem;
  height: 450px;
  width: 300px;
  background-color: rgba(206, 204, 193, 0.9);
  box-sizing: border-box;
  box-shadow: 0px 0px 3px 0px #292727;
  text-align: center;
}

.bike-img,
.bike-name,
.bike-brand,
.bike-model,
.bike-mileage {
  color: #292727;
}

.bike-brand,
.bike-model,
.bike-mileage {
  font-family: 'Lato', sans-serif;
}

.bike-img {
  border: 1px solid #49403c;
  border-radius: 10px 10px 0 0;
  height: 230px;
  width: 296px;
}

.bike-name {
  margin: 15px;
  font-size: 32px;
  font-weight: 700;
  font-family: 'Raleway', sans-serif;
}

.bike-brand {
  margin-bottom: 5px;
  font-size: 24px;
}

.bike-model {
  margin-bottom: 10px;
  font-size: 20px;
  font-style: italic;
  font-weight: 800;
}

.bike-mileage {
  font-size: 24px;
}

.bike-mileage .type {
  font-family: 'Raleway', sans-serif;
  font-size: 24px;
}

@media screen and (max-width: 719px) {
  .bike-card {
    margin: 0 42px 50px 180px;
  }
}

@media screen and (max-width: 610px) {
  .bike-card {
    margin: 0 auto 50px auto;
  }
}

@media screen and (max-width: 450px) {
  .bike-card {
    margin: 0 auto 10px auto;
    height: 410px;
    width: 280px;
  } 

  .bike-img {
    border: 1px solid #49403c;
    border-radius: 10px 10px 0 0;
    height: 230px;
    width: 277px;
  }
}