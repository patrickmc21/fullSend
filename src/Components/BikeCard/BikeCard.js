import React from 'react';
import PropTypes from 'prop-types';
import { convertMetersToMiles } from '../../api/helpers/rideCleaner';

import defaultImg from './images/default.jpeg'
import './BikeCard.css';

const BikeCard = ({bike}) => {
  const distance = convertMetersToMiles(bike.distance);
  const img = bike.description.includes('https://') ? bike.description : defaultImg;
  return (
    <article className='bike-card'>
      <img
        className='bike-img'
        src={img}
        alt='bike image' />
      <h3 className='bike-name'>"{bike.name}"</h3>
      <h4 className='bike-brand'>{bike.brand_name}</h4>
      <h4 className='bike-model'>{bike.model_name}</h4>
      <p className='bike-mileage'>Mileage: {distance}</p>
    </article>
  )
};

export default BikeCard;