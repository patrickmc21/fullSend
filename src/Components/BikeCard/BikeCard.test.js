import React from 'react';
import { shallow } from 'enzyme';
import { mockBikes } from '../../mock-data/mock-data';
import BikeCard from './BikeCard';

describe('BikeCard', () => {

  let wrapper;
  let mockBike;

  beforeEach(() => {
    mockBike = mockBikes[0];
    wrapper = shallow(<BikeCard bike={mockBike}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if user provides pic', () => {
    mockBike.description = 'https://scontent-dfw5-1.xx.fbcdn.net/v/t1.0-9/21';
    wrapper = shallow(<BikeCard bike={mockBike}/>);
    expect(wrapper).toMatchSnapshot();
  });
});