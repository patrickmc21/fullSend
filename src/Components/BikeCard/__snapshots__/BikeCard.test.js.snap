// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikeCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BikeCard
    bike={
        Object {
            "brand_name": "BMC",
            "description": "My Bike.",
            "distance": 388206,
            "frame_type": 3,
            "id": "b1231",
            "model_name": "Teammachine",
            "primary": false,
            "resource_state": 3,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="bike image"
          className="bike-img"
          src="default.jpeg"
/>,
        <h3
          className="bike-name"
>
          "
          "
</h3>,
        <h4
          className="bike-brand"
>
          BMC
</h4>,
        <h4
          className="bike-model"
>
          Teammachine
</h4>,
        <p
          className="bike-mileage"
>
          Mileage: 
          241 miles
</p>,
      ],
      "className": "bike-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "bike image",
          "className": "bike-img",
          "src": "default.jpeg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "\\"",
            undefined,
            "\\"",
          ],
          "className": "bike-name",
        },
        "ref": null,
        "rendered": Array [
          "\\"",
          undefined,
          "\\"",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BMC",
          "className": "bike-brand",
        },
        "ref": null,
        "rendered": "BMC",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Teammachine",
          "className": "bike-model",
        },
        "ref": null,
        "rendered": "Teammachine",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Mileage: ",
            "241 miles",
          ],
          "className": "bike-mileage",
        },
        "ref": null,
        "rendered": Array [
          "Mileage: ",
          "241 miles",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="bike image"
            className="bike-img"
            src="default.jpeg"
/>,
          <h3
            className="bike-name"
>
            "
            "
</h3>,
          <h4
            className="bike-brand"
>
            BMC
</h4>,
          <h4
            className="bike-model"
>
            Teammachine
</h4>,
          <p
            className="bike-mileage"
>
            Mileage: 
            241 miles
</p>,
        ],
        "className": "bike-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "bike image",
            "className": "bike-img",
            "src": "default.jpeg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "\\"",
              undefined,
              "\\"",
            ],
            "className": "bike-name",
          },
          "ref": null,
          "rendered": Array [
            "\\"",
            undefined,
            "\\"",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BMC",
            "className": "bike-brand",
          },
          "ref": null,
          "rendered": "BMC",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Teammachine",
            "className": "bike-model",
          },
          "ref": null,
          "rendered": "Teammachine",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Mileage: ",
              "241 miles",
            ],
            "className": "bike-mileage",
          },
          "ref": null,
          "rendered": Array [
            "Mileage: ",
            "241 miles",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`BikeCard should match the snapshot if user provides pic 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BikeCard
    bike={
        Object {
            "brand_name": "BMC",
            "description": "https://scontent-dfw5-1.xx.fbcdn.net/v/t1.0-9/21761533_10155021148621275_7324564082030573632_n.jpg?_nc_cat=0&oh=bdedd1a008af0a9bf3833e01f4766ef9&oe=5B6FC959",
            "distance": 388206,
            "frame_type": 3,
            "id": "b1231",
            "model_name": "Teammachine",
            "primary": false,
            "resource_state": 3,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="bike image"
          className="bike-img"
          src="https://scontent-dfw5-1.xx.fbcdn.net/v/t1.0-9/21761533_10155021148621275_7324564082030573632_n.jpg?_nc_cat=0&oh=bdedd1a008af0a9bf3833e01f4766ef9&oe=5B6FC959"
/>,
        <h3
          className="bike-name"
>
          "
          "
</h3>,
        <h4
          className="bike-brand"
>
          BMC
</h4>,
        <h4
          className="bike-model"
>
          Teammachine
</h4>,
        <p
          className="bike-mileage"
>
          Mileage: 
          241 miles
</p>,
      ],
      "className": "bike-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "bike image",
          "className": "bike-img",
          "src": "https://scontent-dfw5-1.xx.fbcdn.net/v/t1.0-9/21761533_10155021148621275_7324564082030573632_n.jpg?_nc_cat=0&oh=bdedd1a008af0a9bf3833e01f4766ef9&oe=5B6FC959",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "\\"",
            undefined,
            "\\"",
          ],
          "className": "bike-name",
        },
        "ref": null,
        "rendered": Array [
          "\\"",
          undefined,
          "\\"",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BMC",
          "className": "bike-brand",
        },
        "ref": null,
        "rendered": "BMC",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Teammachine",
          "className": "bike-model",
        },
        "ref": null,
        "rendered": "Teammachine",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Mileage: ",
            "241 miles",
          ],
          "className": "bike-mileage",
        },
        "ref": null,
        "rendered": Array [
          "Mileage: ",
          "241 miles",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="bike image"
            className="bike-img"
            src="https://scontent-dfw5-1.xx.fbcdn.net/v/t1.0-9/21761533_10155021148621275_7324564082030573632_n.jpg?_nc_cat=0&oh=bdedd1a008af0a9bf3833e01f4766ef9&oe=5B6FC959"
/>,
          <h3
            className="bike-name"
>
            "
            "
</h3>,
          <h4
            className="bike-brand"
>
            BMC
</h4>,
          <h4
            className="bike-model"
>
            Teammachine
</h4>,
          <p
            className="bike-mileage"
>
            Mileage: 
            241 miles
</p>,
        ],
        "className": "bike-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "bike image",
            "className": "bike-img",
            "src": "https://scontent-dfw5-1.xx.fbcdn.net/v/t1.0-9/21761533_10155021148621275_7324564082030573632_n.jpg?_nc_cat=0&oh=bdedd1a008af0a9bf3833e01f4766ef9&oe=5B6FC959",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "\\"",
              undefined,
              "\\"",
            ],
            "className": "bike-name",
          },
          "ref": null,
          "rendered": Array [
            "\\"",
            undefined,
            "\\"",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BMC",
            "className": "bike-brand",
          },
          "ref": null,
          "rendered": "BMC",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Teammachine",
            "className": "bike-model",
          },
          "ref": null,
          "rendered": "Teammachine",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Mileage: ",
              "241 miles",
            ],
            "className": "bike-mileage",
          },
          "ref": null,
          "rendered": Array [
            "Mileage: ",
            "241 miles",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
